Add Fluent Bit log-aggregation example for IQ Server HA (CLM-39987) (#69)
sonatype/nexus-iq-server-ha: examples/fluent-bit/README.md

1. IQ Server HA deployed using the Helm chart
2. PVC `iq-server-pvc` exists (default name from chart)
3. Same Kubernetes namespace as IQ Server deployment

## Quick Start

Copy the manifests, edit them to match your environment, then apply:

```bash
# Copy files for editing
cp fluent-bit-configmap.yaml my-fluent-bit-configmap.yaml
cp fluent-bit-daemonset.yaml my-fluent-bit-daemonset.yaml

# Edit both files to set the namespace and PVC claim name to match your IQ
# Server deployment. Defaults assumed by the manifests:
#   namespace:  iq-ha
#   claimName:  iq-server-pvc

# Apply
kubectl apply -f my-fluent-bit-configmap.yaml -f my-fluent-bit-daemonset.yaml

# Verify deployment
kubectl get pods -n <your-namespace> -l app.kubernetes.io/name=fluent-bit
```

## Log Files

Fluent Bit tails these log files from the shared PVC:

| Log Type | Pattern | Description |
|----------|---------|-------------|
| Server | `*-clm-server.log` | Main application logs |
| Request | `*-request.log` | HTTP request logs |
| Audit | `*-audit.log` | Audit trail logs |
| Policy Violation | `*-policy-violation.log` | Policy evaluation logs |
| Stderr | `*-stderr.log` | Standard error output |

## Log Format

> IQ Server logs use this format:

Reviewer: Similar thought to https://github.com/sonatype/nexus-iq-server-ha/pull/69#discussion_r3356908218

Author: 🤖 Same answer as [the chart README thread](https://github.com/sonatype/nexus-iq-server-ha/pull/69#discussion_r3356908218) — the wording was scoped to "server log" in d563c25, and the example README's "Log Formats and Parsers" section now describes each format separately.

### Buffer sizing for large records

Audit and policy-violation records can exceed Fluent Bit's default 32KB
per-line buffer (e.g. an audit `data` payload containing many components, or a
policy-violation event referencing a large dependency tree). The two JSON
inputs set `Buffer_Max_Size` to 1MB so realistic IQ records aren't dropped by
`Skip_Long_Lines`.

### Known limitation: multi-line records

Stack traces and other multi-line entries in `*-clm-server.log` and
`*-stderr.log` fragment because each continuation line is tailed
independently. The first line of an exception parses cleanly; subsequent
`at com.foo.Bar(...)` and `Caused by:` lines fall through to raw
`{"log":"..."}` records. To join continuation lines back to the parent record,
add a `[MULTILINE_PARSER]` to `parsers.conf` and reference it via
`multiline.parser` in the matching `[INPUT]` — see the
[Fluent Bit multiline tail documentation](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/multiline-parsing).

> This was left out of the example to keep the parser configuration legible.

Reviewer: ```suggestion
This was left out of the example for simplicity.
```

Author: 🤖 The original Multi-line records section was rewritten when we corrected the parser name — it now references the custom `iq_java_logback` parser (not the built-in `java`) and explains why the built-in doesn't match IQ Server's comma-millis timestamp format. See commit 3f7e0d3. The original suggestion is superseded by that rewrite.
